test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -246,4 +270,41 @@ UCS_TEST_P(test_ucp_sockaddr, err_handle) {
     EXPECT_EQ(1, err_handler_count);
 }
 
-UCP_INSTANTIATE_TEST_CASE(test_ucp_sockaddr)
+UCP_INSTANTIATE_ALL_TEST_CASE(test_ucp_sockaddr)
+
+class test_ucp_sockaddr_with_rma_atomic : public test_ucp_sockaddr {
+public:
+
+    static ucp_params_t get_ctx_params() {
+        ucp_params_t params = ucp_test::get_ctx_params();
+        params.field_mask  |= UCP_PARAM_FIELD_FEATURES;
+        params.features     = UCP_FEATURE_RMA   |
+                              UCP_FEATURE_AMO32 |
+                              UCP_FEATURE_AMO64;
+        return params;
+    }
+};
+
+UCS_TEST_P(test_ucp_sockaddr_with_rma_atomic, wireup_for_rma_atomic) {
+
+    /* This test makes sure that the client-server flow works when the required
+     * features are RMA/ATOMIC. With these features, need to make sure that
+     * there is a lane for ucp-wireup (an am_lane should be created and used) */
+    struct sockaddr_in connect_addr;
+    get_listen_addr(&connect_addr);
+
+    UCS_TEST_MESSAGE << "Testing " << ucs::sockaddr_to_str((const struct sockaddr*)&connect_addr);
+
+    start_listener((const struct sockaddr*)&connect_addr);
+
+    client_ep_connect((struct sockaddr*)&connect_addr);
+
+    while (receiver().get_num_eps() == 0) {
+        check_events(sender().worker(), receiver().worker(), false);
+    }
+
+    /* allow the connection establishment flow to complete */
+    short_progress_loop();
```

Comment:
why is this needed?

Reply:
How do you ensure that the connection is ready?

Reply:
on the client side we don't have a callback or anything to make sure of that. but this test checks that there is an am_lane for wireup. so for the server side if i remove the forcing of an am_lane, the test will fail on not having an am_lane for wireup - as expected. for client - for connect_to_iface transports, the dont really need an am_lane (for now) since they can connect to the server right away (after getting a wireup_conn_request from it). for connect_to_ep transports, they will have an am_lane regardless of the change in this pr so this test is not intended for testing this.
buttom line, i can remove the forcing of an am_lane for the client and it would still work but we decided to do this anyway since it will probably be needed in the future.